scripts/get_og_previews.py

```diff
@@ -55,9 +57,28 @@ def get_og_preview(url):
     # TODO: Maybe add additional behaviour to check if the image is large enough
     soup = get_page(url)
     image_url = get_og_image(soup)
+
     return image_url
 
 
+def write_image_to_file(url: str, name: str) -> Path | None:
+    hashed = hashlib.shake_128(name.encode("utf-8")).hexdigest(4)
+    path = Path("static") / "previews" / f"{hashed}.jpg"
+
+    # Add Mozilla header to prevent getting blocked for scraping
+    r = httpx.get(url, headers={"User-agent": "Mozilla/5.0"}, follow_redirects=True)
+    if r.status_code != 200:
+        logger.error(f"Couldn't find image at {url}")
+        return None
+
+    logger.info(f"Writing preview {hashed}.jpg for {name}")
+
+    f = open(path, "wb")
+    f.write(r.content)
+    f.close()
+    return Path("previews") / f"{hashed}.jpg"
```

Comment:
I think the assumption of the hosted file being a jpeg isn't a valid one.

Ideally, we actually want to recognize the file type, rescale the image (to something reasonable for a thumbnail), and convert it to a modern file type (`.webp`) to save on load times. I imagine file type recognition can be done through an image library (as opposed to relying on file extension and their various options; e.g., `jpg`, `JPG`, `jpeg`, `JPEG`).

Reply:
Yeah this was actually something I intended to ask, I forgot. Since the formats are compatible it's probably fine to just convert to `.webp` (and taht would preserve `png` transparency). I would actually pass the scaling to Svelte and just store the image as is.